### DISCUSSION TITLE
Ruleset detail view: Add pagination

### DISCUSSION THREAD

**Discussion by Digi92** [mosparo/mosparo#101] [Ideas]
If a user includes larger rulesets in his project, it can quickly become cluttered in the detailed view.
Rulesets can contain many individual rules and the rules themselves can contain many items.
I think there should be some kind of pagination for both cases to keep it more clear.
I could also imagine that particularly large rulesets could generate a memory limit error while rendering the list view.

What is your opinion about this?

**Comment by zepich** [comment]
Hi @Digi92 

You're right. Thank you very much for this great suggestion and for bringing this problem to our attention!

Our test ruleset for the development contains two rules with two rule items each, so we didn't have a problem with that, of course. But the current ruleset detail page is not useful if you have hundreds or thousands of rules and rule items.

I think we should split the views into different pages:

1. You open the ruleset details page. There you see a table (which is sortable and with pagination) with all the rules in the ruleset.
2. You click on one of these rules. You will see a new page where you have a table with all the rule items in this rule, also sortable and with pagination

With this approach, we can make the details page more practical (since you can see all the possible rules in the ruleset), and we can solve a potential memory issue. 

I think we can even add a small filter to filter the kind of rule type in the list of rules. With that, you could filter for only type "X" rules.

We will add that to the next major version.
